Makefile

```
@@ -84,14 +87,22 @@ endif
 # ------
 
 .PHONY: verify
-verify: verify-deps verify-lint verify-licenses
+verify: verify-deps verify-lint verify-licenses verify-gen
 
 .PHONY: verify-deps
 verify-deps:
 	go mod verify
 	go mod tidy
 	@git diff --exit-code -- go.sum go.mod
 
+.PHONY: verify-gen
+verify-gen: # touch go.mod to force generation, then restore its timestamp
+	savetime=$(shell stat -c %Y go.mod); \
+	touch go.mod; \
+	$(MAKE) generate; \
+	touch -d @$${savetime} go.mod
+	@git diff --exit-code -- $(generated_files)
```

Touching the go.mod shouldn't be necessary since without a change made to the go.mod file, the file shouldn't be regenerated and it case the file is modified, it would be caught by the following rule:
```
pkg/generated/openapi/%/zz_generated.openapi.go: go.mod go.sum
```
As such, simplifying to the following rule should be enough:
```suggestion
.PHONY: verify-gen
verify-gen: generate
	@git diff --exit-code -- $(generated_files)
```

---

Yes, it was actually an extra precaution, for this kind of scenario: the OP updates `go.mod`, runs `make`, but **reverts** the changes in the generated files. Then `make verify` does not verify that the generated files are up to date (since their timestamp is greater than `go.mod`'s timestamp).
But I agree that it's a little bit overkill, since the check will be done in CI.

---

In that case we can add the `--always-make` option in CI

---

Ah, thanks, I did not know this option! :pray: 

I've just created a PR in test-infra to use it for `make verify`: https://github.com/kubernetes/test-infra/pull/28412